src/modules/commander/failure_detector/FailureDetector.cpp

```diff
@@ -238,6 +278,26 @@ void FailureDetector::updateAttitudeStatus()
 	}
 }
 
+void FailureDetector::updateFlipStatus()
+{
+
+	rate_ctrl_status_s rate_ctrl_status;
+	//mc_vel_ctrl_status_s mc_vel_ctrl_status;
+	//_sub_mc_vel_ctrl_status.update(&mc_vel_ctrl_status);
+
+	if (_sub_rate_ctrl_status.update(&rate_ctrl_status) && !escaped_z_threshold) {
+
+		bool pr_speed_integ_saturating = (abs(rate_ctrl_status.pitchspeed_integ) >= _param_fd_fail_pri.get()
```

Comment:
> What are you exactly trying to solve?
Early detection of stuck landing gear or failed esc/motors such that the flip itself is prevented.
> Is the "high attitude" detection not working for you? 
I would assume those param values are too high to prevent a flip (That attitude detection is more for reducing the energy of the system before a crash when all hope is lost?) and that check doesn't run during takeoff?
> Monitoring the integrator is also quite tuning-dependent.
yea, I guess it's not easy to tune. I added a param to disable this check. I guess it makes sense to turn it on when flying somewhere muddy or on hills etc.
> What if the cg is not centered and the integrator needs to build-up to compensate for it?
I could add && check for the vz integrator as well? I guess that build up value could also be used to estimate a normalised take off weight? I could also think of using a moment of inertia...